pkg/imageprovider/rhcos.go

```diff
@@ -84,8 +93,38 @@ func imageKey(data imageprovider.ImageData) string {
 	)
 }
 
+func getIgnitionOverride(imageMetadata *metav1.ObjectMeta, log logr.Logger) ([]byte, error) {
+	if overrideURI, exist := imageMetadata.Annotations[ignitionOverrideAnnotation]; exist {
+		log.Info("using Ignition override when building the image", "host", imageMetadata.Name, "overrideURI", overrideURI)
+		resp, err := http.Get(overrideURI) //#nosec G107
```

Comment:
Does this support TLS? Anything we need to do to enable that?

Reply:
https://issues.redhat.com/browse/METAL-155

Reply:
METAL-155 is about providing images via TLS IIUC. I think Flavio's question is about *receiving* the Ignition.

Reply:
@zaneb  correct! AI will be serving these through TLS, just want to make sure we won't run into issues. We should also have a way to disable TLS verification or, as @dtantsur said, make sure the CA is properly mounted.

Reply:
On my environment:
```yaml
  volumes:
  - ...
  - name: kube-api-access-cqplg
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
      - configMap:
          items:
          - key: service-ca.crt
            path: service-ca.crt
          name: openshift-service-ca.crt
```